src/charm.py

```diff
@@ -197,54 +272,41 @@ def update(self, event):
             event.defer()
             return
 
-        self.model.unit.open_port(port=APPLICATION_PORT, protocol="tcp")
+        charm_function = self.config["charm-function"].value
+        logger.info(f"configuring ranger {charm_function}")
 
-        if self.config.get(
-            "user-group-configuration"
-        ) and self.unit.status == ActiveStatus("Status check: UP"):
-            self.group_manager._handle_synchronize_file(event)
+        if charm_function == "admin":
+            self.model.unit.open_port(port=APPLICATION_PORT, protocol="tcp")
+            command, context = self._configure_ranger_admin(container)
 
-        logger.info("configuring ranger")
-        db_conn = self._state.database_connection
-        context = {
-            "DB_NAME": db_conn["dbname"],
-            "DB_HOST": db_conn["host"],
-            "DB_PORT": db_conn["port"],
-            "DB_USER": db_conn["user"],
-            "DB_PWD": db_conn["password"],
-            "RANGER_ADMIN_PWD": self.config["ranger-admin-password"],
-            "JAVA_OPTS": "-Duser.timezone=UTC0",
-        }
-
-        config = render("config.jinja", context)
-        container.push(
-            "/usr/lib/ranger/install.properties", config, make_dirs=True
-        )
+        if charm_function == "usersync":
+            command, context = self._configure_ranger_usersync(container)
 
-        logger.info("planning ranger execution")
+        logger.info(f"planning ranger {charm_function} execution")
```

Comment:
Lazy strings are recommended for Python logging, i.e., `"planning ranger %s execution", charm_function`